addons/website_sale_collect/static/src/js/payment_form.js

```diff
@@ -0,0 +1,37 @@
+import { _t } from '@web/core/l10n/translation';
+import PaymentForm from '@payment/js/payment_form';
+
+PaymentForm.include({
+    /**
+     * Keep payment button context to switch labels displayed on payment method change.
+     *
+     * @override
+     */
+    async start() {
+        await this._super(...arguments);
+        const buttons = document.querySelectorAll('button[name="o_payment_submit_button"]');
+        this.paymentButtonContext = {
+            buttonLabelMapping: [...buttons].map((btn) => [btn, btn.textContent]),
+            payOnSiteLabel: _t("Confirm"),
+        };
+    },
+
+    /**
+     * Set a custom payment button label when `pay_on_site` method gets selected.
+     * Restore original label if method changes away from `pay_on_site`.
+     *
+     * @override
+     */
+    _selectPaymentOption(ev) {
+        this._super.apply(this, arguments);
+        const pmCode = this._getPaymentMethodCode(ev.target);
+        const { buttonLabelMapping, payOnSiteLabel } = this.paymentButtonContext;
+        buttonLabelMapping.forEach(pmCode === 'pay_on_site'
+            ? ([button, _]) => button.textContent = payOnSiteLabel
+            : ([button, label]) => {
+                if (button.textContent === payOnSiteLabel) {
+                    button.textContent = label;
+                }
+            });
```

Comment:
seems simpler? we can even extract the condition in a separate variable `isPayOnsite = pmCode === 'pay_on_site'` if it is beneficial
```suggestion
        buttonLabelMapping.forEach(([button, label]) => {
            button.textContent = pmCode === 'pay_on_site'? payOnSiteLabel : label;
        });
```

Reply:
Oh right, another concern was that if a third option were to be added, it would always get overwritten by the original label if any PM other than `pay_on_site` were to be selected. With the current iteration, the original label only gets restored when changing away from `pay_on_site`.

> if you want to avoid unnecessary writes, maybe we can use a normal loop to make it more readable

I'm more partial to the succinct version, but I can see your point, and pushed a different version